os/include/tinyara/mm/mm.h

```diff
@@ -396,6 +396,10 @@ struct mm_heap_s {
 	int mm_nregions;
 #endif
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	size_t elf_data_size;
```

Comment:
Since, elf_data_size consists of both text and data sections, I think its better to rename this variable as elf_sections_size or elf_areas_size.